### ISSUE TITLE
TypedDict.update does not work properly.

### ISSUE THREAD

**Issue by zhou13** [microsoft/pyright#10789]
**Is your feature request related to a problem? Please describe.**
```
from typing import TypedDict


class A(TypedDict):
    a: int


class B(TypedDict):
    a: int
    b: int


a1: A = {"a": 0}
b1: B = {"a": 0, "b": 0}

b1.update(a1)
```

I got error 
```
Argument of type "A" cannot be assigned to parameter "__m" of type "Partial[B]" in function "update"
  "b" is an incompatible type
    "object" is not assignable to "int"Pylance[reportArgumentType](https://github.com/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md)
```

Similarly, 

```python
b1 = b1 | a1
```
also does not work with error
```
Type "A" is not assignable to declared type "B"
  "b" is missing from "A"Pylance[reportAssignmentType](https://github.com/microsoft/pylance-release/blob/main/docs/diagnostics/reportAssignmentType.md)
```

**Describe the solution you’d like**
A clear and concise description of the solution you would like to see.

I think this should work without error.

**Comment by erictraut** [comment]
Allowing `update` to work in this case would be unsafe from a type perspective. Consider the following:

```python
class C(A):
    b: str

a1: A
a1 = C(a=1, b="")
b1: B = {"a": 0, "b": 0}

b1.update(a1)
```

This would result in a value that purports to have type `B`, but it would have a `str` value associated with its `b` key.

In any case, pyright is simply honoring the type information provided in typeshed here — for both the `update` and `__or__` methods. For details, refer to the class definition for `TypedDictFallback` [here](https://github.com/python/typeshed/blob/6cae34647c8fa41060cd9de5a7de3ce6239b6b19/stdlib/_typeshed/_type_checker_internals.pyi#L38). If you'd like to make the case for relaxing the definitions for `update` or `__or__`, you can file an issue in the [typeshed issue tracker](https://github.com/python/typeshed/issues) or post in the [Python typing forum](https://discuss.python.org/c/typing/32). Historically, the typing community has fallen more on the side of strictness when it comes to `TypedDict`, but you might be able to make the case for relaxing it here — trading off some false positives for more potential false negatives.

For now, I'm going to close this issue because I think pyright's arguably doing the right thing, and it's consistent with other type checkers.
